rust/src/linalg/cosine.rs

```diff
@@ -60,6 +93,22 @@ impl Cosine for [f32] {
     }
 }
 
+impl Cosine for [f64] {
```

Comment:
n00b question: why does this have to be done multiple times? is it not possible to use a type parameter?

Reply:
This was due that rust does not support generic specialization, which can not allow us to write different code for different type. 
We need to write type-specific SIMD (i.e,  bf16. and f64 uses different SIMD routines). These are placeholders.